### PR TITLE
Add S.fromEither to pull out values from either

### PR DESCRIPTION
**Pull request by wennergr**
S.fromEither :: b -> Either a b -> b

Closes: #130

### CODE REVIEW

**Comment by davidchambers** [comment]
Thank you, @wennergr! This was a nice surprise. :)

**Comment by davidchambers** [comment]
The order in which the functions appear in the source file determines the order in which the functions are listed in the documentation, so it's nice to group related functions. In this case, let's follow the (somewhat arbitrary) precedent set by `fromMaybe`—which appears after `isJust` and before `maybe`—by inserting `fromEither` between `isRight` and `either`.

File: test/fromEither.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+describe('fromEither', function() {
+
+  it('is a binary function', function() {
+    eq(typeof S.fromEither, 'function');
+    eq(S.fromEither.length, 2);
+  });
+
+  it('type checks its arguments', function() {
+    throws(function() { S.fromEither(0, [1, 2, 3]); },
+                   'Invalid value\n' +
```
Comment (davidchambers) [inline test/fromEither.js]: Let's adjust the indentation like so:

```javascript
throws(function() { S.fromEither(0, [1, 2, 3]); },
       'Invalid value\n' +
```

File: test/fromEither.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                   'The value at position 1 is not a member of ‘Either a b’.\n');
+  });
+
+  it('can be applied to a Left', function() {
+    eq(S.fromEither(0, S.Right(42)), 42);
+  });
+
+  it('can be applied to a Right', function() {
+    eq(S.fromEither(0, S.Left(42)), 0);
+  });
```
Comment (davidchambers) [inline test/fromEither.js]: The expressions and the accompanying descriptions should be switched. ;)

**Comment by wennergr** [comment]
Thanks for your feedback

I did not do property based tests because the fromMaybe tests didn't. Is it only used to test laws? Performance reasons? More then happy to change ;)

Also more then happy to rebase before the merge.

**Comment by davidchambers** [comment]
> I did not do property based tests because the fromMaybe tests didn't. Is it only used to test laws?

The truthful answer is that property-based testing is still relatively new to me, so I don't think to use it. While it would be great to add more property-based tests, that's a larger change orthogonal to this one.

> Also more then happy to rebase before the merge.

That would be marvellous. :)

I'd like to add that I :heart: the way in which you went about this pull request. You matched the existing code and documentation style, and referenced the issue this pull request will close. Very nice!

**Comment by wennergr** [comment]
Done, and thanks!

**Comment by davidchambers** [comment]
:deciduous_tree:

**Comment by davidchambers** [comment]
```
$ npm install sanctuary@0.12.0

$ node
> const S = require('sanctuary')
undefined
> S.fromEither
fromEither :: b -> Either a b -> b
```

**Pull request merged**